File: day17/src/main/java/de/havox_design/aoc2017/day17/Day17.java
package de.havox_design.aoc2017.day17;

import de.havox_design.aoc.utils.DataReader;

import java.util.List;

public class Day17 {
    private final List<String> input;

    public Day17(String fileName) {
        input = readData(fileName);
    }

    public static long solvePart1(String fileName) {
        Day17 instance = new Day17(fileName);
        return instance.solvePart1();
    }

    public static long solvePart2(String fileName) {
        Day17 instance = new Day17(fileName);
        return instance.solvePart2();
    }

    public long solvePart1() {
        return 0L;
    }

    public long solvePart2() {return 0L;
    }

    private List<String> readData(String fileName) {
        return DataReader.readData(fileName, MainClass.class);
    }
}

File: templates/scala/aoc20xx-scala/src/main/scala/de/havox_design/aoc20xx/day17/Day17.scala
package de.havox_design.aoc2022.day17

object Day17 {
  def solvePart1(filename: String): Long = {
    1L
  }

  def solvePart2(filename: String): Long = {
    1L
  }

  def main(args: Array[String]): Unit = {
    println("Solution for part1: " + solvePart1("de/havox_design/aoc20xx/day17/day17.txt"))
    println("Solution for part2: " + solvePart2("de/havox_design/aoc20xx/day17/day17.txt"))
  }

  private def readData(filename: String): Iterator[String] =
    scala
      .io
      .Source
      .fromResource(filename)
      .getLines()
}
